python/cuml/dask/ensemble/base.py

```diff
@@ -81,18 +90,20 @@ def _concat_treelite_models(self):
         to create a single model. The concatenated model is then converted to
         bytes format.
         """
-        mod_bytes = []
+        model_protobuf_futures = list()
         for w in self.workers:
-            mod_bytes.append(self.rfs[w].result().model_pbuf_bytes)
+            model_protobuf_futures.append(
+                dask.delayed(_get_protobuf_bytes)
+                (self.rfs[w]))
+        mod_bytes = self.client.compute(model_protobuf_futures, sync=True)
         last_worker = w
         all_tl_mod_handles = []
         model = self.rfs[last_worker].result()
-        for n in range(len(self.workers)):
-            all_tl_mod_handles.append(model._tl_model_handles(mod_bytes[n]))
-        concat_model_handle = model._concatenate_treelite_handle(
-            treelite_handle=all_tl_mod_handles)
+        all_tl_mod_handles = [model._tl_model_handles(pbuf_bytes)

```

Comment:
Nice. Centralizing all this code will make it easier to debug the current memory work.